### PR TITLE
fix: row border in flex layout should overflow

### LINKED ISSUE #27570: [Bug]: DataGrid / Table: Row border does not extend horizontally when rows overflow horizontally
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
"@fluentui/react-components": "9.18.6"
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/s/fluent-ui-v9-datagrid-h13r52?file=/example.tsx

## Actual Behavior
Recording_20230414T164059
image

## Expected Behavior
The row borders should extend horizontally.

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by ling1726**
Currently row border using flex layout in the Table should overflow with cell content. Adds `min-width: fit-content` to flex layout styles of the `Table` so that the cells don't overflow the Table

Fixes #27570

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/InfoButton-mount_base.html">12</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/Avatar-mount_base.html">585</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/Avatar-mount.html">587</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/Button-mount_base.html">298</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/Button-mount.html">306</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/Field-mount_base.html">1013</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/Field-mount.html">1021</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/FluentProvider-mount_base.html">677</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/FluentProvider-mount.html">643</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/FluentProviderWithTheme-mount_base.html">79</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/FluentProviderWithTheme-mount.html">83</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">74</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">71</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">75</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/InfoButton-mount_base.html">12</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/MakeStyles-mount_base.html">860</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/MakeStyles-mount.html">852</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/Persona-mount_base.html">1615</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/Persona-mount.html">1572</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/SpinButton-mount_base.html">1294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28161/perf-test-react-components/SpinButton-mount.html">1289</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                     |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                  Change |
| :-------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                         | `150.941 kB`<br />`41.535 kB` | `151.053 kB`<br />`41.572 kB` | `112 B` <br />`37 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>    |  `45.147 kB`<br />`12.577 kB` |  `45.259 kB`<br />`12.613 kB` | `112 B` <br />`36 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>         | `133.392 kB`<br />`34.008 kB` | `133.504 kB`<br />`34.048 kB` | `112 B` <br />`40 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr> |  `79.161 kB`<br />`19.386 kB` |   `79.273 kB`<br />`19.42 kB` | `112 B` <br />`34 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>           |  `78.491 kB`<br />`19.195 kB` |  `78.603 kB`<br />`19.229 kB` | `112 B` <br />`34 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `64.935 kB`<br />`17.918 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `211.151 kB`<br />`58.715 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.168 kB`<br />`11.964 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.473 kB`<br />`2.196 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e101115deef04a486083afb56af060c03d9fa9f3'>e101115deef04a486083afb56af060c03d9fa9f3</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-table-bb208d47-ce79-49d3-bb79-3b635f0d5e95.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-table-bb208d47-ce79-49d3-bb79-3b635f0d5e95.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 3 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=28161&commitId=d4315e0d0d5c3652b62feacb6d003d4512e59fab&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1941 screenshots unchanged, 0 screenshots with different dimensions and 3 screenshots with visible difference.


  <details><summary><b>unknown</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Table layout flex - truncate.default (disabled).chromium.png|11830| Changed|
|Table layout flex - truncate.false.chromium.png|11830| Changed|
|Table layout flex - truncate.true.chromium.png|6919| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-table/src/components/TableRow/useTableRowStyles.styles.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@ const useFlexLayoutStyles = makeStyles({
  */
 const useStyles = makeStyles({
   root: {
+    // width: 'max-content',
```
Comment (Hotell) [inline packages/react-components/react-table/src/components/TableRow/useTableRowStyles.styles.ts:32]: forgot to remove ?
```suggestion
```

File: packages/react-components/react-table/src/components/Table/useTableStyles.styles.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ const useTableLayoutStyles = makeStyles({
 const useFlexLayoutStyles = makeStyles({
   root: {
     display: 'block',
+    minWidth: 'fit-content',
```
Comment (Hotell) [inline packages/react-components/react-table/src/components/Table/useTableStyles.styles.ts:23]: might be a good idea to add docs here as well ( which is part of PR description ) . do we have this covered by browser tests ?

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Comment by layershifter** [comment]
Closing for housekeeping 🏡  Feel free to re-open 👍

**Pull request closed without merging**